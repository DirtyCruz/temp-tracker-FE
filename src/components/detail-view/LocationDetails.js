import React from 'react';
import PropTypes from 'prop-types';
import LocationDetail from './LocationDetail';

function LocationDetails({ location, temps }) {
  const tempsList = temps.map((temp, i) => (
    <li key={i}>
      <LocationDetail temp={temp} />
    </li>
  ));
  return (
    <>
    <h3>{location}</h3>
    <ul>
      {tempsList}
    </ul>
    </>
  );
}

LocationDetails.propTypes = {
  location: PropTypes.string.isRequired,
  temps: PropTypes.array.isRequired
};

export default LocationDetails;
