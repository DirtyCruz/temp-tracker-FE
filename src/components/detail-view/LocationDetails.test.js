import React from 'react';
import { shallow } from 'enzyme';
import LocationDetails from './LocationDetails';

describe('LocationDetails component', () => {
  it('renders LocationDetails', () => {
    const wrapper = shallow(<LocationDetails location='Portland' temps={[4, 5, 7]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
