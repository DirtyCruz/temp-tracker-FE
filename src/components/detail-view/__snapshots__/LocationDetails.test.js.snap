// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetails component renders LocationDetails 1`] = `
<Fragment>
  <h3>
    Portland
  </h3>
  <ul>
    <li
      key="0"
    >
      <LocationDetail
        temp={4}
      />
    </li>
    <li
      key="1"
    >
      <LocationDetail
        temp={5}
      />
    </li>
    <li
      key="2"
    >
      <LocationDetail
        temp={7}
      />
    </li>
  </ul>
</Fragment>
`;
